packages/contracts/contracts/test/CommonTest.t.sol

```diff
@@ -93,9 +93,89 @@ contract CommonTest is Test {
+        return Proxy(payable(target));
+    }
+}
+
+contract UpgradeGovernor_Initializer is CommonTest {
+    address superAdmin = makeAddr("superAdmin");
+    address owner = makeAddr("owner");
+
+    // Constructor arguments
+    uint256 internal initialVotingDelay = 0;
+    uint256 internal initialVotingPeriod = 30;
+    uint256 internal initialProposalThreshold = 1;
+    uint256 internal votesQuorumFraction = 70;
+    uint256 internal minDelaySeconds = 3;
+    string internal baseUri = "";
+
+    // Test data
+    address internal guardian1 = 0x0000000000000000000000000000000000001004;
+    address internal guardian2 = 0x0000000000000000000000000000000000001005;
+    address internal guardian3 = 0x0000000000000000000000000000000000001006;
+    address internal notGuardian = 0x0000000000000000000000000000000000001006;
+
+    address[] timeLockProposers = new address[](1);
+    address[] timeLockExecutors = new address[](1);
+
+    SecurityCouncilToken securityCouncilToken = SecurityCouncilToken(address(new Proxy(multisig)));
+    TimeLock timeLock = TimeLock(payable(address(new Proxy(multisig))));
+    UpgradeGovernor upgradeGovernor = UpgradeGovernor(payable(address(new Proxy(multisig))));
+
+    SecurityCouncilToken securityCouncilTokenImpl = new SecurityCouncilToken();
+    TimeLock timeLockImpl = new TimeLock();
+    UpgradeGovernor upgradeGovernorImpl = new UpgradeGovernor();
+
+    function setUp() public virtual override {
+        super.setUp();
+        vm.startPrank(multisig);
+
+        // setup SecurityCouncilToken
+        pp(address(securityCouncilToken)).upgradeToAndCall(
+            address(securityCouncilTokenImpl),
+            abi.encodeCall(SecurityCouncilToken.initialize, owner)
+        );
+
+        // setup TimeLock & UpgradeGovernor
+        timeLockProposers[0] = address(upgradeGovernor);
+        timeLockExecutors[0] = address(upgradeGovernor);
+
+        pp(address(timeLock)).upgradeToAndCall(
+            address(timeLockImpl),
+            abi.encodeCall(
+                TimeLock.initialize,
+                (minDelaySeconds, timeLockProposers, timeLockExecutors, address(upgradeGovernor))
+            )
+        );
+
+        pp(address(upgradeGovernor)).upgradeToAndCall(
+            address(upgradeGovernorImpl),
+            abi.encodeCall(
+                UpgradeGovernor.initialize,
+                (
+                    address(securityCouncilToken),
+                    payable(address(timeLock)),
+                    initialVotingDelay,
+                    initialVotingPeriod,
+                    initialProposalThreshold,
+                    votesQuorumFraction
+                )
+            )
+        );
+
+        //change proxy admin to upgradeGovernor
+        pp(address(securityCouncilToken)).changeAdmin(address(upgradeGovernor));
+        pp(address(timeLock)).changeAdmin(address(upgradeGovernor));
+        pp(address(upgradeGovernor)).changeAdmin(address(upgradeGovernor));
```

Comment:
Should the admins of these contracts be changed to timeLock? not upgradeGovernor.